secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+typealias ZEC = NSDecimalNumber
+
+// MARK: - Zatoshi
+
+struct Zatoshi {
+    @Clamped(-(100_000_000 * 21_000_000)...(100_000_000 * 21_000_000))
+    var amount: Int64 = 0
+
+    /// Converts `Zatoshi` to `ZEC`
+    var asZec: ZEC {
+        ZEC(value: Double(amount) / Double(100_000_000))
+    }
+    
+    /// The value of `Zatoshi` is valid when zero or positive number. User can't send or receive negative amount of `Zatoshi`.
+    /// Negative `Zatoshi` makes sense only as some partial result, generally speaking during some computations.
+    /// Another example would be summing up last 10 transactions resulting in negative `Zatoshi` value.
+    var isValid: Bool {
+        amount >= 0
+    }
+    
+    static func + (left: Zatoshi, right: Zatoshi) -> Zatoshi {
+        Zatoshi(amount: left.amount + right.amount)
+    }
+
+    static func - (left: Zatoshi, right: Zatoshi) -> Zatoshi {
+        Zatoshi(amount: left.amount - right.amount)
+    }
+}
+
+extension Int {
+    var asZatoshi: Zatoshi { Zatoshi(amount: Int64(self)) }
+    var asZec: ZEC { ZEC(value: Double(self)) }
+}
+
+extension Int64 {
+    var asZatoshi: Zatoshi { Zatoshi(amount: self) }
+    var asZec: ZEC { ZEC(value: Double(self)) }
+}
+
+// MARK: - ZEC
+
+extension ZEC {
+    static var nsDecimalHandler = NSDecimalNumberHandler(
+        roundingMode: ZEC.RoundingMode.plain,
```

Comment:
I believe the expected rounding mode should be Bankers' 
see: https://github.com/zcash/secant-android-wallet/blob/e44e9016769450c4e312c25992ad73d3d123936e/sdk-ext-ui-lib/src/main/java/cash/z/ecc/sdk/ext/ui/model/ZecString.kt#L90

Reply:
bankers it is